core/string/string_name.h

```diff
@@ -124,6 +124,12 @@ class StringName {
 	static StringName search(const char32_t *p_name);
 	static StringName search(const String &p_name);
 
+	_FORCE_INLINE_ int length() const {
+		return this->operator String().length();
```

Comment:
`operator String()` can be an expensive operation. I'm not sure the impact when the data is stored as a `String`, it may not be a problem there. But when the data is a static C string, `operator String()` will create a new `String` from that every time. I don't hiding something like that into a seemingly quick function like `length()`.